driver/utils/type_info.h

```diff
@@ -1555,6 +1564,13 @@ namespace value_manip {
             static inline void convert(const SourceType & src, DestinationType & dest) {
                 return from_value<decltype(src.value)>::template to_value<DestinationType>::convert(src.value, dest);
             }
+            // Allow further optimizations down the road by stealing value from `src` with either move() or swap().
+            static inline void convert(SourceType & src, DestinationType & dest) {
+                return from_value<decltype(src.value)>::template to_value<DestinationType>::convert(src.value, dest);
+            }
+            static inline void convert(SourceType && src, DestinationType & dest) {
+                return from_value<decltype(src.value)>::template to_value<DestinationType>::convert(src.value, dest);
```

Comment:
I guess it should be `std::move(src.value)` here. Also, not every `DataSourceType<Id>` has `.value` member.